src/libbson/doc/bson_oid_compare_unsafe.rst

```diff
@@ -0,0 +1,9 @@
+:man_page: bson_oid_compare_unsafe
+
+bson_oid_compare_unsafe()
+=========================
+Description
+-----------
+
+Identical to :symbol:`bson_oid_compare()`, but does not check that its arguments are non-null.
```

Comment:
```diff
- Identical to :symbol:`bson_oid_compare()`, but does not check that its arguments are non-null.
+ Identical to :symbol:`bson_oid_compare()`, but performs no integrity checking.
```
For consistency with (some) `bson_iter_*_unsafe()` [comment documentation](https://github.com/mongodb/mongo-c-driver/blob/2baa45ca752ffadd57ed8ea167fbe633db7e044a/src/libbson/src/bson/bson-iter.h#L146) which expands the scope to all preconditions, not just not-null preconditions.

Suggestion applies to other instances in this PR.